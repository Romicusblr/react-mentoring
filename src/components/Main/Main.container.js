import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router-dom';
import { fetchMovies } from '../../actions';
import Main from './Main';

const mapStateToProps = ({ movies }) => ({
  data: movies,
  loading: movies.loading,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchMovies,
}, dispatch);

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Main));
