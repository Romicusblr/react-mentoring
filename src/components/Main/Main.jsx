import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import SearchBody from '../SearchBody';
import MainHeader from '../MainHeader';
import style from './Main.module.css';

const Main = (props) => {
  const {
    data, className, fetchMovies, ...other
  } = props;

  useEffect(() => {
    fetchMovies();
  }, []);

  if (!data) return null;
  return (
    <main
      className={classNames(style.main, className)}
      {...other}
    >
      <MainHeader quantity={data.total} />
      <SearchBody movies={data.data} />
    </main>
  );
};

Main.propTypes = {
  data: PropTypes.shape({
    data: PropTypes.arrayOf(PropTypes.objectOf),
    total: PropTypes.number,
  }),
  className: PropTypes.string,
  fetchMovies: PropTypes.func.isRequired,
};

Main.defaultProps = {
  className: '',
  data: null,
};

export default Main;
