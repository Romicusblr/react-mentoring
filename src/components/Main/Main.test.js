import Main from '.';

fdescribe('Main', () => {
  it('should be selectable by class "main"', () => {
    expect(shallow(<Main />).is('.main')).toBe(true);
  });

  it('should match to snapshot', () => {
    const snap = shallow(<Main />);
    expect(snap).toMatchSnapshot();
  });
});
