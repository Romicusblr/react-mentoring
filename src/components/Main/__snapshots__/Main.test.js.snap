// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match to snapshot 1`] = `
<main
  className="main"
>
  <MainHeader
    className=""
  />
  <SearchBody
    className=""
    movies={Array []}
  />
</main>
`;
